perun-ldapc/src/main/java/cz/metacentrum/perun/ldapc/model/impl/PerunResourceImpl.java

```diff
@@ -76,13 +77,15 @@
 				);
 	}
 
-	public void addResource(Resource resource, String entityID) throws InternalErrorException {
+	public void addResource(Resource resource, Map<String,String> facilityAttributes) throws InternalErrorException {
 		addEntry(resource);
 		// get info about entityID attribute if exists
-		if(entityID != null) {
+		if (!facilityAttributes.isEmpty()) {
 			try {
 				DirContextOperations context = findByDN(buildDN(resource));
-				context.setAttributeValue(PerunAttribute.PerunAttributeNames.ldapAttrEntityID, entityID);
+				for (String ldapAttrName : facilityAttributes.keySet()) {
+					context.setAttributeValue(ldapAttrName, facilityAttributes.get(ldapAttrName));
+				}
```

Comment:
I think this is wrong. You should use ldap attribute names as configured in XML context and not suppose they are the same as their Perun counterparts.

Reply:
I no longer have specific addResource() method, I use sync operation instead.